user/timepix3/module/src/Timepix3Event2StdEventConverter.cc

```diff
@@ -81,8 +82,53 @@ bool Timepix3TrigEvent2StdEventConverter::Converting(eudaq::EventSPC ev, eudaq::
 uint64_t Timepix3RawEvent2StdEventConverter::m_syncTime(0);
 uint64_t Timepix3RawEvent2StdEventConverter::m_syncTime_prev(0);
 size_t Timepix3RawEvent2StdEventConverter::m_clearedHeader(0);
+bool Timepix3RawEvent2StdEventConverter::m_first_time(true);
+bool Timepix3RawEvent2StdEventConverter::applyCalibration(false);
+std::string Timepix3RawEvent2StdEventConverter::calibrateDetector("");
+std::string Timepix3RawEvent2StdEventConverter::calibrationPath("");
+std::string Timepix3RawEvent2StdEventConverter::threshold("");
+std::vector<std::vector<float>> Timepix3RawEvent2StdEventConverter::vtot;
+std::vector<std::vector<float>> Timepix3RawEvent2StdEventConverter::vtoa;
+
 bool Timepix3RawEvent2StdEventConverter::Converting(eudaq::EventSPC ev, eudaq::StandardEventSP d2, eudaq::ConfigurationSPC conf) const{
 
+  // Read from configuration:
+  if(m_first_time) {
+      if(conf->Has("calibrate_detector") && conf->Has("calibration_path") && conf->Has("threshold")) {
+          calibrateDetector = conf->Get("calibrate_detector","");
+          calibrationPath = conf->Get("calibration_path","");
+          threshold = conf->Get("threshold","");
+          applyCalibration = true;
+          EUDAQ_INFO("Applying calibration from " + calibrationPath + " for detector " + calibrateDetector);
+
+          // make paths to calibration files and read
+            std::string tmp;
+            tmp = calibrationPath + "/" + calibrateDetector + "/cal_thr_" + threshold + "_ik_10/" + calibrateDetector + "_cal_tot.txt";
```

Comment:
this is already hacky in the Timepix3 event loader, please don't inherit that style. If you pass a "file name" then this should be a canonical path already.

Reply:
Fair enough, it's simplified now.